sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseDataCollector.java

```diff
@@ -413,4 +448,84 @@ static CountAndDuration decodeCountAndDuration(long countAndDuration) {
             return new CountAndDuration(countAndDuration >> 44, countAndDuration & MAX_DURATION);
         }
     }
+
+    class OTelMetricsStorage {
+        private volatile ConcurrentHashMap<String, OTelMetric> metrics = new ConcurrentHashMap<>();
+        // setting most amount of OTel metrics a user can stream to this. will review if users surpass threshold.
+        private final int maxMetricsLimit = 50;
+
+        public void addMetric(String metricName, double value) {
+            OTelMetric metric = metrics.get(metricName);
+            if (metric == null) {
+                // create new metric and add to metrics map
+                // (if we have reached the limit, we will block the metric from being created
+                if (metrics.size() <= maxMetricsLimit) {
+                    metric = new OTelMetric(metricName);
+                    metric.addDataPoint(value);
+                    metrics.putIfAbsent(metricName, metric);
+                }
+            } else {
+                metric.addDataPoint(value);
+            }
+        }
+
+        public ArrayList<QuickPulseMetrics> processMetrics() {
+            ConcurrentHashMap<String, ArrayList<QuickPulseConfiguration.DerivedMetricInfo>> requestedMetrics = quickPulseConfiguration.getDerivedMetrics();
+            ArrayList<QuickPulseMetrics> processedMetrics = new ArrayList<>();
+
+            for(Map.Entry<String, OTelMetric> entry : this.metrics.entrySet()) {
+                String key = entry.getKey();
+                OTelMetric value = entry.getValue();
+
+                if (requestedMetrics.containsKey(key)) {
+                    for (QuickPulseConfiguration.DerivedMetricInfo metricInfo : requestedMetrics.get(key)) {
+                        if (Objects.equals(metricInfo.getTelemetryType(), "Metric")) {
+                            QuickPulseMetrics processedMetric = processMetric(value, metricInfo);
+                            processedMetrics.add(processedMetric);
+                        }
+                    }
+                }
+            }
+
+            this.clearMetrics();
+            return processedMetrics;
+
+        }
+
+        public QuickPulseMetrics processMetric(OTelMetric metric, QuickPulseConfiguration.DerivedMetricInfo metricInfo) {
```

Comment:
After your demo, looking at this piece of code again - it looks like within the scope of a single second, the code expects that a specific metric "x" might have multiple values (hence the list of data values), and performs aggregations on those multiple values. Considering that you mentioned there is only one sampling of an otel metric in a second, does it make sense to use the Aggregation field here to perform some aggregation? It might be good to clarify the intent for how Aggregation should be used in the case of the Metric telemetry type here.